src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -174,13 +183,21 @@ export default class WorkspaceContainer extends Component {
           ]),
         h(ClusterManager, { namespace })
       ]),
-      h(WorkspaceTabs, { namespace, name, activeTab, refresh, workspace, onDelete: this.onDelete }),
+      h(WorkspaceTabs, {
+        namespace, name, activeTab, refresh, workspace,
+        onDelete: this.onDelete, onClone: this.onClone
+      }),
       div({ style: { position: 'relative', flexGrow: 1 } }, [
         this.props.children
       ]),
       deletingWorkspace && h(DeleteWorkspaceModal, {
         namespace, name,
         onDismiss: () => this.setState({ deletingWorkspace: false })
+      }),
+      cloningWorkspace && h(NewWorkspaceModal, {
+        cloneWorkspace: workspace,
+        onCreate: ({ namespace, name }) => Nav.goToPath('workspace', { namespace, name }),
```

Comment:
It's probably safe to bake this behavior into the component.

Reply:
Agree